File: app/src/main/java/org/neidhardt/dynamicsoundboard/soundsheetmanagement/views/ConfirmDeleteAllSoundSheetsDialog.java
package org.neidhardt.dynamicsoundboard.soundsheetmanagement.views;

import android.app.FragmentManager;
import org.neidhardt.dynamicsoundboard.R;
import org.neidhardt.dynamicsoundboard.dao.SoundSheet;
import org.neidhardt.dynamicsoundboard.mediaplayer.EnhancedMediaPlayer;
import org.neidhardt.dynamicsoundboard.soundmanagement_old.ServiceManagerFragment;
import org.neidhardt.dynamicsoundboard.soundmanagement_old.model.SoundDataModel;
import org.neidhardt.dynamicsoundboard.views.BaseConfirmDeleteDialog;

import java.util.List;

/**
 * File created by eric.neidhardt on 16.02.2015.
 */
public class ConfirmDeleteAllSoundSheetsDialog extends BaseConfirmDeleteDialog
{
	private static final String TAG = ConfirmDeleteAllSoundSheetsDialog.class.getName();

	public static void showInstance(FragmentManager manager)
	{
		ConfirmDeleteAllSoundSheetsDialog dialog = new ConfirmDeleteAllSoundSheetsDialog();
		dialog.show(manager, TAG);
	}

	@Override
	protected int getInfoTextResource()
	{
		return R.string.dialog_confirm_delete_all_soundsheets_message;
	}

	@Override
	protected void delete()
	{
		List<SoundSheet> soundSheets = this.soundSheetsDataAccess.getSoundSheets();
		this.getSoundActivity().removeSoundFragments(soundSheets);

		SoundDataModel model = ServiceManagerFragment.getSoundDataModel();
		for (SoundSheet soundSheet : soundSheets)
		{
			List<EnhancedMediaPlayer> soundsInSoundSheet = model.getSoundsInFragment(soundSheet.getFragmentTag());
			model.removeSounds(soundsInSoundSheet);
		}

		this.soundSheetsDataStorage.removeAllSoundSheets();
	}
}

File: app/src/main/kotlin/org/neidhardt/dynamicsoundboard/soundsheetmanagement/views/ConfirmDeleteAllSoundSheetsDialog.kt
package org.neidhardt.dynamicsoundboard.soundsheetmanagement.views

import android.support.v4.app.FragmentManager
import org.neidhardt.dynamicsoundboard.R
import org.neidhardt.dynamicsoundboard.SoundboardApplication
import org.neidhardt.dynamicsoundboard.views.BaseConfirmDeleteDialog

/**
 * File created by eric.neidhardt on 16.02.2015.
 */
class ConfirmDeleteAllSoundSheetsDialog : BaseConfirmDeleteDialog()
{
	private val soundSheetsDataAccess = SoundboardApplication.soundSheetsDataAccess
	private val soundSheetsDataStorage = SoundboardApplication.soundSheetsDataStorage
	private val soundsDataAccess = SoundboardApplication.soundsDataAccess
	private val soundsDataStorage = SoundboardApplication.soundsDataStorage

	override val infoTextResource: Int = R.string.dialog_confirm_delete_all_soundsheets_message

	override fun delete()
    {
		val soundSheets = this.soundSheetsDataAccess.getSoundSheets()
		this.soundActivity.removeSoundFragments(soundSheets)

		for (soundSheet in soundSheets)
        {
			val soundsInSoundSheet = this.soundsDataAccess.getSoundsInFragment(soundSheet.fragmentTag)
			this.soundsDataStorage.removeSounds(soundsInSoundSheet)
		}
		this.soundSheetsDataStorage.removeSoundSheets(soundSheets)
	}

	companion object
    {
		private val TAG = ConfirmDeleteAllSoundSheetsDialog::class.java.name

		fun showInstance(manager: FragmentManager) {
			val dialog = ConfirmDeleteAllSoundSheetsDialog()
			dialog.show(manager, TAG)
		}
	}
}
